packages/toolpad-app/src/toolpadComponents/index.tsx

```diff
@@ -3,7 +3,9 @@ import * as appDom from '../appDom';
 
 export interface ToolpadComponentDefinition {
   displayName: string;
-  builtin?: string;
+  builtIn?: string;
+  display: boolean;
+  displayIcon?: string;
```

Comment:
Just `icon` should be fine 👍 

```suggestion
  icon?: string;
```

Reply:
In fact, I'd just implement this with a Map that maps component name to an icon component and not add a `icon` property at all